reference/best_practices/reusable_bundles.rst

```diff
@@ -1,4 +1,98 @@
 Reusable Bundles for Akeneo PIM community
 =========================================
 
-Coming soon!!! :)
+You are an integrator and you want to share your project with the awesome Akeneo PIM community?
+Let's read our advices to make a reusable and documented bundle!
+
+
+What do you want when you create a reusable bundle?
+
+* **Compatibility**: to fit Akeneo PIM features
+* **Maintainability**: to make your code flexible and testable (and tested)
+* **Extensibility**: to allow integrators to use your bundle to fit with their specific needs
+* **Scalability**: to know your application limits and document them
+
+
+Compatibility
+-------------
+
+Your bundle won't be reusable for some customers if you forgot some cases. That is no more an issue if you document compatibilities of your bundle.
+
+Akeneo PIM storage stack comes with MySQL and hybrid MySQL/MongoDB (:doc:`/reference/technical_information/choose_database`).
+Use the ProductQueryBuilder which provides an abstraction layer making both storages work with the same API.
+You can refer to this cookbook for more details and use cases: ":doc:`/cookbook/catalog/product/query`".
+
+Two existing distributions CE/EE. The main project is `Akeneo PIM Community Edition`_ (CE) but
+we also have an `Enterprise Edition`_ (EE) which comes with a couple of extra features dedicated to advanced needs.
+
+Many attribute types are managed (14 in CE and one more in EE). Make sure you do not forget any use case for an attribute type.
+
+Try your bundle on other catalogs. Don't forget use cases with many locales, many channels and various catalog volumes.
+You can use our `data generator bundle`_ to generate a catalog with random data.
+
+.. _Akeneo PIM Community Edition: https://github.com/akeneo/pim-community-standard
+.. _Enterprise Edition: https://www.akeneo.com/enterprise-edition/
+.. _data generator bundle: https://github.com/akeneo-labs/DataGeneratorBundle
+
+
+Maintainability
+---------------
+
+First, keep in mind the `Single Responsibility`_ from the five `SOLID principles`_ of Oriented Object Programming.
+It seems simple but we often forget to explode our code in many thinner classes.
+The code is easily readable and testable.
+
+You can use `PMD`_ to help you to make your code simple, flexible and easily testable.
+
+
+Now your code should be testable, so test your code! :)
+In Akeneo PIM, we use `PhpSpec`_ as unit testing tool and `Behat`_ to test our application.
+
+Use a Continuous Integration to deploy your application, test your code and improve its quality.
```

Comment:
Imho, we can advise, share what we do, but these are not mandatory rules. A guy can release a very good extension without writing TDD, using CI, phpspec, behat, cs-fixer, etc 

If the guy properly use the PIM API and avoid to override everything, the code will be maintainable even without a single line of test.